### DISCUSSION TITLE
A plan for PocketBase plugins

### DISCUSSION THREAD

**Discussion by benallfree** [pocketbase/pocketbase#3679] [Q&A]
Taking another shot at this.. :)

I'm envisioning a WordPress-like plugin system where a plugin is a zip file that contains:

* Svelte frontend bundle that integrates with the admin
* pb_hooks
* migrations (up/down)
* version compatibility markers

And then maybe there can be a `pocketbase-plugins` codex/repo a la DefinitelyTyped where devs can fork and contrib and maybe eventually some day update their own plugins.

Now that I am more familiar with golang, I would be willing to lend some horsepower to this effort. 

What do you think @ganigeorgiev?

Plugin ideas already suggested:

* bulk mail
* nightly data reset
* stats
* stripe/lemonsqueezy webhook endpoint and integration

**Comment by lightningspirit** [comment]
Divide to conquer... I think this idea is clever for the next generation of contributions on Pocketbase.

**Comment by gokulgseven** [comment]
I would assume we can use something like plugins, by creating a library of the (migrations, pb_hooks (jsvm apis)  all packaged in a zip file), 
basically, a set of templates/instructions on which fields to create and the apis to go with that functionality. 

if we can get a iframe display in admin ui , we can also create the plugin front end with htmx(all packaged in pb_hooks) and no worry about the svelte integration.

it wont really be plugins (plugins) , but it will be something closer. As a guess the plugins part is way ahead in the roadmap.

**Comment by pchasco** [comment]
What sort of use cases are you imagining for PocketBase plugins?

**Reply by gokulgseven** [reply]
something like integrating stripe quickly, and other such services, quality of life things that just make coding certain features faster

**Comment by babakfp** [comment]
### Plugin Idea

A plugin that can do this:

https://github.com/pocketbase/pocketbase/discussions/1914

Maybe useful info: It seems like PocketBase can restart itself:

https://github.com/pocketbase/pocketbase/issues/4324#issuecomment-1937034482

> on backup restore we restart the app in the background

**Comment by moussa32** [comment]
### Plugin Idea

A plugin that controls a corn job and fetches endpoints

**Comment by ganigeorgiev** [comment]
I'm closing this discussion as it starting to become too spammy.

Support for Admin UI plugins will be added in the future but that will be after the Admin UI rewrite and the internal refactoring as there will be some breaking changes to the exposed Go/JS interfaces.

**Reply by kuwts** [reply]
Thanks for this update, I just discovered Pocketbase and haven't been this excited about a new tool/framework in a long time, so much potential here.
